synapse/handlers/identity.py

```diff
@@ -175,11 +179,19 @@ def unbind_threepid(self, mxid, threepid):
             content=content,
             destination_is=id_server,
         )
-        yield self.http_client.post_json_get_json(
-            url,
-            content,
-            headers,
-        )
+        try:
+            yield self.http_client.post_json_get_json(
+                url,
+                content,
+                headers,
+            )
+        except HttpResponseException as e:
+            if e.code in (400, 404, 501,):
+                # The remote server probably doesn't support unbinding (yet)
+                defer.returnValue(False)
+            else:
+                raise SynapseError(502, "Failed to contact identity server")
```

Comment:
shouldn't we include some info about the error we got from the IS?

Reply:
Yeah, possibly. Not sure what we'd want to copy? @richvdh did you have opinions about how this is meant to work?

Reply:
err, not really. It feels like it's probably not something a user is going to do much about, so I'm not sure there is much value in including it in the response.

What we probably *should* do is log the error.